README
NAME
    Dist::Zilla::PluginBundle::TestingMania - test your dist with every
    testing plugin conceivable

VERSION
    version 0.010

SYNOPSIS
    In dist.ini:

        [@TestingMania]

DESCRIPTION
    This plugin bundle collects all the testing plugins for Dist::Zilla
    which exist (and are not deprecated). This is for the most paranoid
    people who want to test their dist seven ways to Sunday.

    Simply add the following near the end of dist.ini:

        [@TestingMania]

    It includes the most recent version (as of release time) of the
    following plugins, in their default configuration. Note that not all the
    plugins are actually *used* by default.

  Testing plugins
    *   Dist::Zilla::Plugin::CompileTests, which performs tests to syntax
        check your dist.

    *   Dist::Zilla::Plugin::ConsistentVersionTest, which tests that all
        modules in the dist have the same version. See
        Test::ConsistentVersion for details. This is not enabled by default;
        see "Enabling Tests".

    *   Dist::Zilla::Plugin::CriticTests, which checks your code against
        best practices. See Perl::Critic for details.

    *   Dist::Zilla::Plugin::DistManifestTests, which tests MANIFEST for
        correctness. See Test::DistManifest for details.

    *   Dist::Zilla::Plugin::EOLTests, which ensures the correct line
        endings are used (and also checks for trailing whitespace). See
        Test::EOL for details.

    *   Dist::Zilla::Plugin::HasVersionTests, which tests that your dist has
        version numbers. See Test::HasVersion for what that means.

    *   Dist::Zilla::Plugin::KwaliteeTests, which performs some basic
        kwalitee checks. *Kwalitee* is an automatically-measurable guage of
        how good your software is. It bears only a superficial resemblance
        to the human-measurable guage of actual quality. See Test::Kwalitee
        for a description of the tests.

    *   Dist::Zilla::Plugin::MetaTests, which performs some extra tests on
        META.yml. See Test::CPAN::Meta for what that means.

    *   Dist::Zilla::Plugin::Test::CPAN::Meta::JSON, which performs some
        extra tests on META.json, if it exists. See Test::CPAN::Meta::JSON
        for what that means.

    *   Dist::Zilla::Plugin::MinimumVersionTests, which tests for the
        minimum required version of perl. See Test::MinimumVersion for
        details, including limitations.

    *   Dist::Zilla::Plugin::NoTabsTests, which ensures you don't use *The
        Evil Character*. See Test::NoTabs for details. If you wish to
        exclude this plugin, see "Disabling Tests".

    *   Dist::Zilla::Plugin::PodCoverageTests, which checks that you have
        Pod documentation for the things you should have it for. See
        Test::Pod::Coverage for what that means.

    *   Dist::Zilla::Plugin::PodSyntaxTests, which checks that your Pod is
        well-formed. See Test::Pod and perlpod for details.

    *   Dist::Zilla::Plugin::PortabilityTests, which performs some basic
        tests to ensure portability of file names. See
        Test::Portability::Files for what that means.

    *   Dist::Zilla::Plugin::SynopsisTests, which does syntax checking on
        the code from your SYNOPSIS section. See Test::Synopsis for details
        and limitations.

    *   Dist::Zilla::Plugin::UnusedVarsTests, which checks your dist for
        unused variables. See Test::Vars for details.

    *   Dist::Zilla::Plugin::Test::CPAN::Changes, which checks your
        changelog for conformance with CPAN::Changes::Spec. See
        Test::CPAN::Changes for details.

        Set "changelog" in dist.ini if you don't use Changes:

            [@TestingMania]
            changelog = CHANGELOG

  Disabling Tests
    To exclude a testing plugin, give a comma-separated list in dist.ini:

        [@TestingMania]
        disable = EOLTests,NoTabsTests

  Enabling Tests
    This pluginbundle may have depend on some testing plugins that aren't
    enabled by default. This option allows you to turn them on. Attempting
    to add plugins which are not listed above will have *no effect*.

    To enable a testing plugin, give a comma-separated list in dist.ini:

        [@TestingMania]
        enable = ConsistentVersionTest

AVAILABILITY
    The project homepage is
    <http://p3rl.org/Dist::Zilla::PluginBundle::TestingMania>.

    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <http://search.cpan.org/dist/Dist-Zilla-PluginBundle-TestingMania/>.

    The development version lives at
    <http://github.com/doherty/Dist-Zilla-PluginBundle-TestingMania> and may
    be cloned from
    <git://github.com/doherty/Dist-Zilla-PluginBundle-TestingMania.git>.
    Instead of sending patches, please fork this project using the standard
    git and github infrastructure.

SOURCE
    The development version is on github at
    <http://github.com/doherty/Dist-Zilla-PluginBundle-TestingMania> and may
    be cloned from
    <git://github.com/doherty/Dist-Zilla-PluginBundle-TestingMania.git>

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <https://github.com/doherty/Dist-Zilla-PluginBundle-TestingMania/issues>
    .

AUTHOR
    Mike Doherty <doherty@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Mike Doherty.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



CHANGES
Changelog for Dist::Zilla::PluginBundle::TestingMania

{{$NEXT}}

0.008     2011-04-16
 * Include a test for META.json - disable it if your dists don't include this
 * Removed ProgCriticTests (was disabled by default)
 * Removed CheckChangesTests
 * Documentation improvements

0.007     2011-04-12
 * New prereq names

0.006     2011-04-04
 * Set default changelog; thanks to mjgardner

0.005     2011-04-02
 * Added ChangesTests

0.004     2011-02-15
 * Removed PodLinkTests; the dependencies are too immature

0.003     2011-01-31
 * Removed ApacheTests
 * Removed PodSpellingTests, which has a fail in the dependency chain, wasn't
   enabled by default, and is too annoying to bear.
 * Added PodLinkTests

0.002     2010-11-28
 * Flesh out the chain of dependencies to get full functionality
 * Use Perl_5 license

0.001     2010-10-18
 * Initial version


Makefile.PL

use strict;
use warnings;

BEGIN { require 5.010001; }

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'test your dist with every testing plugin conceivable',
  'AUTHOR' => 'Mike Doherty <doherty@cpan.org>',
  'BUILD_REQUIRES' => {
    'File::Find' => '0',
    'File::Path' => '0',
    'File::Temp' => '0',
    'Moose::Autobox' => '0',
    'Test::DZil' => '0',
    'Test::More' => '0.88',
    'autodie' => '0'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.31'
  },
  'DISTNAME' => 'Dist-Zilla-PluginBundle-TestingMania',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'Dist::Zilla::PluginBundle::TestingMania',
  'PREREQ_PM' => {
    'Dist::Zilla::Plugin::CompileTests' => '0',
    'Dist::Zilla::Plugin::ConsistentVersionTest' => '0',
    'Dist::Zilla::Plugin::CriticTests' => '1.102280',
    'Dist::Zilla::Plugin::DistManifestTests' => '0',
    'Dist::Zilla::Plugin::EOLTests' => '0.02',
    'Dist::Zilla::Plugin::HasVersionTests' => '0',
    'Dist::Zilla::Plugin::KwaliteeTests' => '0',
    'Dist::Zilla::Plugin::MetaTests' => '0',
    'Dist::Zilla::Plugin::MinimumVersionTests' => '0',
    'Dist::Zilla::Plugin::NoTabsTests' => '0',
    'Dist::Zilla::Plugin::PodCoverageTests' => '0',
    'Dist::Zilla::Plugin::PodSyntaxTests' => '0',
    'Dist::Zilla::Plugin::PortabilityTests' => '0',
    'Dist::Zilla::Plugin::SynopsisTests' => '0',
    'Dist::Zilla::Plugin::Test::CPAN::Changes' => '0',
    'Dist::Zilla::Plugin::Test::CPAN::Meta::JSON' => '0.003',
    'Dist::Zilla::Plugin::Test::Pod::LinkCheck' => '0',
    'Dist::Zilla::Plugin::UnusedVarsTests' => '0',
    'Dist::Zilla::Role::PluginBundle::Easy' => '0',
    'Moose' => '0',
    'Pod::Coverage::TrustPod' => '0',
    'Test::CPAN::Meta' => '0',
    'Test::Perl::Critic' => '0'
  },
  'VERSION' => '0.010',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



